File: src/Base7.java
public class Base7 {
    public static String convertToBase7(int number) {
        if (number == 0) return "0";
        StringBuilder result = new StringBuilder();
        boolean isNegative = number < 0;
        if (isNegative) {
            number *= -1;
        }
        while (number > 0) {
            result.append(number % 7);
            number /= 7;
        }
        return result.append(isNegative ? "-" : "").reverse().toString();
    }

    public static void main(String[] args) {
        System.out.println(convertToBase7(-7));
    }
}

File: python/base_7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return '0'
        is_negative = num < 0
        num = -num if is_negative else num
        result = ''
        while num > 0:
            result = str(num % 7) + result
            num //= 7
        return ('-' if is_negative else '') +  result
